pkg/microservice/systemconfig/core/codehost/handler/codehost.go

```diff
@@ -128,38 +129,52 @@ func Callback(c *gin.Context) {
 	defer func() { internalhandler.JSONResponse(c, ctx) }()
 
 	state := c.Query("state")
+	state, err := url.PathUnescape(state)
+	if err != nil {
+		ctx.Logger.Errorf("code_host_get_call_back PathUnescape state:%s err : %s", state, err)
+		url := fmt.Sprintf("%s%s", state, "&errMessage=failed_to_parse_redirect_url")
```

Comment:
what if state does not have other query strings, for example: state="http://example.com"